fastlane/Fastfile

```diff
@@ -122,7 +122,7 @@ platform :android do
   desc "Upload a snapshot release"
   lane :deploy_snapshot do |options|
     version = android_get_version_name(gradle_file: gradle_file_path)
-    deploy unless is_snapshot_version?(version)
+    deploy unless !is_snapshot_version?(version)
```

Comment:
if `is_snapshot_version` is `true`, the negation will actually deploy it right? Unless executes code if conditional is false.

Reply:
I think this is fine. What happened is that it actually wasn't a SNAPSHOT version (we hadn't updated develop).

Since we are not checking for the version in the `config.yaml` in the `deploy-snapshot` job, the lane will execute, and we will try to deploy.

I think we should just have one lane `deploy`. Then have two jobs in CircleCI, one for snapshots, that gets only triggered in develop and checks that the version number is actually a snapshot number, and one for deploys that get triggered on tags.

Reply:
yeah which is what we want here. the code used to be:
`if !is_snapshot_version then return end
    deploy`

so we want to deploy when it IS snapshot version, since this is the upload snapshot release lane

Reply:
You're right

Reply:
that's so opposite of how i would think....

Reply:
wait yes so this is what we want. we want it to deploy only snapshot versions.

so if is_snapshot_version is true, !is_snapshot_version is false and it will deploy.

Reply:
Totally lol, so confusing. I think if we are negating, the boolean, we can do if else, otherwise is like a double negation.

Reply:
Yes, I see it now haha: "Deploy unless it is not a snapshot version"

Reply:
sorry your other comment didn't load before i commented...i'm fine with having the two jobs. do you want to do that or should i?

Reply:
ok, I totally lost track of what we want and what we don't, but quick note about code simplicity: this should either be

```ruby
deploy unless is_snapshot_version?(version)

```

or 

```ruby
deploy if is_snapshot_version?(version)
```

Reply:
> deploy if is_snapshot_version?(version)
Is what we want